File: TSF/Scripts/TSF/Data/SortElement.ts
namespace TSF.Data {
    /** sort direction of a query */
    export enum SortDirection {
        ASC,DESC
    }
    /**
     * The sort element of a query.
     */
    export class SortElement {
        /** the column to sort by */
        public column:string ;
        /** the sort direction */
        public sortDirection: SortDirection;
        /**
         * The sort element of a query.  
         * @param column - the column to sort by 
         * @param sortDirection - the direction to sort by (optional.  Sorts by Ascending by default).
         */
        constructor(column: string, sortDirection: SortDirection = SortDirection.ASC)
        {
            this.column = column;
            this.sortDirection = sortDirection;
        }
    }
}

File: TSF.CSharp/Data/SortElement.cs
namespace TSF.CSharp.Data
{
    /// <summary>
    /// sort direction of a query
    /// </summary>
    public enum SortDirection
    {
        ASC, DESC
    }
    /// <summary>
    /// The sort element of a query.
    /// </summary>
    public class SortElement
    {
        /// <summary>
        /// the column to sort by
        /// </summary>
        public string column;
        /// <summary>
        /// the sort direction
        /// </summary>
        public SortDirection sortDirection;

        /// <summary>
        /// The sort element of a query.
        /// </summary>
        /// <param name="column">the column to sort by </param>
        /// <param name="sortDirection">the direction to sort by (optional.  Sorts by Ascending by default).</param>

        SortElement(string column, SortDirection sortDirection = SortDirection.ASC)
        {
            this.column = column;
            this.sortDirection = sortDirection;
        }

        SortElement()
        {

        }
    }
}